### PR TITLE
Add YOLO26 semantic segmentation export support

### PR DESCRIPTION
**Pull request by rolandocortez**
## Purpose

Add support for exporting YOLO26 semantic segmentation models from `tools`.

This enables `yolo26n-sem.pt` to be exported as an ONNX NNArchive with a semantic segmentation head configuration, so the resulting archive can be converted through HubAI and used directly with `ParsingNeuralNetwork` / `SegmentationParser`.

## Specification

This PR adds:

* `yolov26-sem` version detection.
* Routing for `yolov26-sem` in the main conversion flow.
* Semantic segmentation mode in `Yolo26Exporter`.
* Added a reusable `make_segmentation_nn_archive()` helper in `Exporter` to generate NNArchives with `SegmentationParser` + `HeadSegmentationMetadata`.
* E2E and semantic-specific test coverage for the generated NNArchive head.

The YOLO26 semantic segmentation checkpoint is detected through the `SemanticSegment` head in the model checkpoint.

The generated archive keeps the existing YOLO26 output naming convention and uses `output_yolo26` as the semantic segmentation output.

The generated ONNX output is:

```text
output_yolo26 [1, 19, 416, 416]
```

The generated NNArchive head config is:

```text
parser: SegmentationParser
outputs: ["output_yolo26"]
metadata:
  n_classes: 19
  is_softmax: false
```

No changes were needed in `depthai-nodes`, since the existing `SegmentationParser` already supports a single semantic segmentation logits output and produces a `SegmentationMask`.

I also verified that the existing YOLO26 detect / instance segmentation / pose output-name behavior stays unchanged.

## Dependencies & Potential Impact

No new dependencies.

Impact should be limited to YOLO26 semantic segmentation export. Existing YOLO26 export paths are not intended to change.

## Deployment Plan

No special deployment required.

Rollback is reverting this PR.

## Testing & Validation

Tests:

```bash
pytest tests/test_end2end.py::test_yolo26_semseg_nnarchive_head --download-weights -s
# 1 passed

pytest tests/test_end2end.py --download-weights --test-case yolo26n-sem -s
# 2 passed, 102 skipped
```

Manual export validation:

* Exported `yolo26n-sem.pt` through `tools`.
* Verified the ONNX output name and shape: `output_yolo26 [1, 19, 416, 416]`.
* Verified the generated NNArchive uses `SegmentationParser`.
* Verified the head outputs are `["output_yolo26"]`.
* Verified the metadata has 19 classes and `is_softmax=false`.

Hub conversion validation:

* RVC4 FP16: PASS
* RVC4 INT8: PASS
* RVC2: PASS

The converted archive configs were inspected and confirmed to preserve:

```text
parser: SegmentationParser
outputs: ["output_yolo26"]
metadata:
  n_classes: 19
  is_softmax: false
```

Runtime validation:

* RVC4 FP16: PASS
* RVC4 INT8: PASS
* RVC2: PASS

The converted archives were tested with a DepthAI pipeline using:

```text
ReplayVideo -> ParsingNeuralNetwork -> ApplyColormap -> ImgFrameOverlay
```

The parser output was:

```text
depthai_nodes.message.segmentation.SegmentationMask
```

and the visualizer rendered semantic segmentation masks / overlays correctly.

I also validated the RVC2 archive locally with `oak-examples/neural-networks/generic-example` on OAK-1 using `-overlay`.

## AI Usage

Assisted-by: ChatGPT

Submitted code was reviewed by a human: YES

The author is taking responsibility for the contribution: YES

## Release Notes

* **New Features**
  * Added support for YOLOv26 semantic-segmentation variants, including automatic detection.
  * Extended the CLI/export flow to generate semantic-segmentation NN archives with the correct head and metadata.

* **Tests**
  * Added an end-to-end test covering NN archive export and semantic-segmentation head configuration for the YOLOv26 semantic-segmentation variant.

### CODE REVIEW

File: tools/modules/exporter.py [hunk 1]
```diff
@@ -243,6 +244,73 @@ def make_nn_archive(
         )
         archive.make_archive()
 
+    def make_segmentation_nn_archive(
```
Comment (klemen1999) [inline tools/modules/exporter.py]: Let's rename to `make_semantic_seg_nn_archive` to make it more obvious what this is doing

**Review by klemen1999 — COMMENTED** [review]
Overall LGTM, just one nitpick and general question:
Do we potentially need any ultralytic submodule version bump? Just checking...

**Comment by rolandocortez** [comment]
> Overall LGTM, just one nitpick and general question: Do we potentially need any ultralytic submodule version bump? Just checking...

I don't think we need a bumping right now. The current submodule pointer already includes SemanticSegment, and the targeted yolo26n-sem NNArchive test passes with the existing submodule version

**Review by klemen1999 — APPROVED** [review]
